workers/worker_base.py

```diff
@@ -894,3 +897,33 @@ def update_gh_rate_limit(self, response, bad_credentials=False, temporarily_disa
 
             # Change headers to be using the new oauth's key
             self.headers = {'Authorization': 'token %s' % self.oauths[0]['access_token']}
+
+    def init_oauth_gitlab(self):
+        self.oauths = []
+        self.headers = None
+
+        # Endpoint to hit solely to retrieve rate limit information from headers of the response
+        url = "https://gitlab.com/api/v4/version"
+
+        # Make a list of api key in the config combined w keys stored in the database
+        oauthSQL = s.sql.text("""
+                    SELECT * FROM worker_oauth WHERE access_token <> '{}' and platform = 'gitlab'
+                """.format(self.config['gitlab_api_key']))
+
+        for oauth in [{'oauth_id': 0, 'access_token': self.config['gitlab_api_key']}] + json.loads(pd.read_sql(oauthSQL, self.helper_db, params={}).to_json(orient="records")):
+            self.headers = {'Authorization': 'Bearer %s' % oauth['access_token']}
+            logging.info("Getting rate limit info for oauth: {}\n".format(oauth))
+            response = requests.get(url=url, headers=self.headers)
+            self.oauths.append({
+                    'oauth_id': oauth['oauth_id'],
+                    'access_token': oauth['access_token'],
+                    'rate_limit': int(response.headers['ratelimit-remaining']),
+                    'seconds_to_reset': (datetime.datetime.fromtimestamp(
+                        int(response.headers['ratelimit-reset'])) - datetime.datetime.now()).total_seconds()
```

Comment:
unlike github, there's a pretty nifty header attribute  `'RateLimit-ResetTime'`.
This would remove the need to convert it into a datetime object. 
You could directly use response.headers['RateLimit-ResetTime'] - datetime.datetime.now()

Reply:
Not sure of this .. but I think this is it !